### PR TITLE
[OCPCLOUD-1679] Rewrite rebase bot to use merge-tmp to create base branch

### PR DESCRIPTION
**Pull request by JoelSpeed**
This rewrites the bot to set up the rebase branch before applying the cherry picks using the tried and tested method, based on the instructions below (taken from the latest autoscaler rebase):
```
$ git remote update # make sure we update our refs
$ git checkout cluster-autoscaler-1.25.0
$ git checkout -b merge-tmp # create a temporary branch for our merge commit
$ git checkout openshift/master # we want to be at the tip of the openshift master branch when we run the next command
$ echo 'merge upstream/cluster-autoscaler-1.25.0' | git commit-tree merge-tmp^{tree} -p HEAD -p merge-tmp -F - # create a new merge commit for our history
deadbeef12345678 # id of new merge commit
$ git branch merge-1.25.0 deadbeef12345678 # create a new branch for the cherry-pick work
$ git checkout merge-1.25.0
$ git cherry-pick <carry commits> # cherry pick the needed commits into the new branch
```
With this, we now set up the merge history, drop any carries that we already have, and then reapply the carries to the worktree, which should result in fewer conflicts and issues going forward

### CODE REVIEW

File: rebasebot/bot.py [hunk 1]
```diff
@@ -161,37 +161,31 @@ def _do_rebase(gitwd, source, dest, source_repo, tag_policy):
...
-        gitwd.git.checkout(f"dest/{dest.branch}")
-        gitwd.git.merge("--no-ff", "rebase")
-    except git.GitCommandError:
-        logging.info("Merge commit is required")
-        return True
-    finally:
-        gitwd.git.reset("--hard", "HEAD")
-        gitwd.git.checkout("rebase")
+        gitwd.git.branch("-d", "merge-tmp", force=True)
+    except:
```
Comment (lobziik) [inline rebasebot/bot.py:170]: Do not use bare except, please... can you please specify what exception you trying to catch here?

File: rebasebot/bot.py [hunk 2]
```diff
@@ -161,37 +161,31 @@ def _do_rebase(gitwd, source, dest, source_repo, tag_policy):
...
-        gitwd.git.merge(
-            f"dest/{dest.branch}", "-Xours", "-m",
-            f"UPSTREAM: <carry>: Merge branch '{dest.branch}' in {gitwd.active_branch}"
-        )
-    except git.GitCommandError as ex:
-        if not _resolve_conflict(gitwd):
-            logging.info("Merge conflict has been automatically resolved.")
-            raise RepoException(f"Git merge failed: {ex}") from ex
+        gitwd.git.branch("-d", "rebase", force=True)
+    except:
```
Comment (lobziik) [inline rebasebot/bot.py:185]: Same as previous comment, specify exception type please

File: rebasebot/bot.py [hunk 3]
```diff
@@ -161,37 +161,31 @@ def _do_rebase(gitwd, source, dest, source_repo, tag_policy):
...
     try:
-        gitwd.git.checkout(f"dest/{dest.branch}")
-        gitwd.git.merge("--no-ff", "rebase")
-    except git.GitCommandError:
-        logging.info("Merge commit is required")
-        return True
-    finally:
-        gitwd.git.reset("--hard", "HEAD")
-        gitwd.git.checkout("rebase")
+        gitwd.git.branch("-d", "merge-tmp", force=True)
```
Comment (lobziik) [inline rebasebot/bot.py:169]: Nit: i thinking about constant for this "merge-tmp". Not necessary, but would look a bit better, imho

**Review by lobziik — COMMENTED** [review]
I did check results in cloud-provider-azure, looked good to me, thanks a lot for this!

couple of nits

**Review by elmiko — COMMENTED** [review]
the changes generally make sense to me, i concur with @lobziik 's comments about cleanup

**Comment by JoelSpeed** [comment]
Hmm, need to get you added to the collaborators, I'll merge manually for now

**Pull request merged**